src/amo/components/GetFirefoxBanner/index.js

```diff
@@ -57,7 +57,8 @@ export const GetFirefoxBannerBase = ({
   const onButtonClick = () => {
     _tracking.sendEvent({
       action: GET_FIREFOX_BANNER_CLICK_ACTION,
-      category: getDownloadCategory(variant),
+      category: GET_FIREFOX_BUTTON_CLICK_CATEGORY,
+      extra: variant,
```

Comment:
The overall approach looks both good and simple and I think it will work (I haven't tested the patch yet) but the "interface" (or API) is not great.

We should be more explicit here because no one will expect to see a second event sent by simply passing an `extra` prop. The JSDoc isn't going to help because no one expect documented functions either. I think we should be more verbose. Here is an example:

```js
_tracking.sendEvent({
  action: GET_FIREFOX_BANNER_CLICK_ACTION,
  category: getDownloadCategory(variant),
  category: GET_FIREFOX_BUTTON_CLICK_CATEGORY,
  sendSecondEventWithOverrides: variant && {
    category: `${GET_FIREFOX_BUTTON_CLICK_CATEGORY}-${variant}`,
  }
});
```

and then the logic in `sendEvent` would remain pretty much the same:

```js
if (typeof sendSecondEventWithOverrides === 'object') {
  const secondEvent = { ...data, ...sendSecondEventWithOverrides };
  this._ga('send', extraData);
  this.log('sendEvent', extraData);
}
```

Reply:
Sure, your suggestion is definitely more explicit, and also allows us to override the second event in any way we choose. I'll make that change.